docs/core-concepts/app-deployment.md

```diff
@@ -1,17 +1,255 @@
+                 Console.ReadKey();
+          }
+
+          private static void ShowWords(String s)
+          {
+              String pattern = @"\w+";
+              var matches = Regex.Matches(s, pattern);
+              if (matches.Count == 0)
+                  Console.WriteLine("\nNo words were identified in your input.");
+              else {
+                  Console.WriteLine("\nThere are {0} words in your string:", matches.Count);
+                  for (int ctr = 0; ctr < matches.Count; ctr++)
+                      Console.WriteLine("   #{0,2}: '{1}' at position {2}", ctr,
+                                        matches[ctr].Value, matches[ctr].Index);
+              }
+          } 
+      }
+    }
+    ```
+
+3. Run the `dotnet restore` command to restore the dependencies specified in your project.
+
+4. Create a debug build of your app by using the `dotnet build` command.
+
+5. After you've debugged and tested the program, you can create the files to be deployed with your app by using the `dotnet publish -c release` command. This creates a release (rather than a debug) version of your app.
+
+   The resulting files are placed in a directory named `publish` that is in a subdirectory of your project's
+`.\bin\release\netcoreapp1.0` subdirectory.
+
+6. Along with your application's files, the publishing process emits a program database (.pdb) file that contains debugging information about your app. You should not package it with your application's files.
+
+The remaining files can be deployed in any way you'd like. For example, you can package them in a zip file, use a simple `copy` command, or deploy them with any installation package of your choice.
+
+## Deploying a Portable App with Third-Party Dependencies ##
+
+Deploying a portable app with one or more third-party dependencies involves two additional steps before you can run the `dotnet restore` command:
+
+1. Add references to any third-party libraries to the `dependencies` section of your `project.json` file. The following  `dependencies` section uses Json.NET as a third-party library.
+
+
+    ```
+    "dependencies": {
+      "Microsoft.NETCore.App": {
+        "type": "platform",
+        "version": "1.0.0"
+      },
+      "Newtonsoft.Json": "8.0.3"
+    },
+    ```
+
+2. Ensure that you have installed the NuGet package that contains the third-party dependency, since third-party dependencies are resolved from your local NuGet cache.
+
+Note that a portable app with third-party dependencies will only be as portable as its third-party dependencies. For example, if a third-party library only supports OSX, the app will not be portable to Windows systems.
+
+When you deploy your application, any third-party dependencies used in your app are resolved from the local NuGet cache of the system on which your app is running. In other words, the third-party library must be present on the target machine for your app to run successfully.

```

Comment:
When I run `dotnet publish` with `"Newtonsoft.Json": "9.0.1-beta1"` in project.json (I couldn't make 8.0.3 work), `Newtonsoft.Json.dll` (and its dependencies) is copied to the `publish` directory. Shouldn't that be deployed and then used to run the application?